servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/server/subsystems/Karaf2xStartupLaunchConfigurator.java

```diff
@@ -191,10 +182,36 @@ protected String getVMArguments(String karafInstallDir) {
 		return vmArguments.toString();
 	}
 
+	/**
+	 * @param karafInstallDir
+	 * @param vmLoc
+	 * @return
+	 */
+	String createEndorsedDirValue(String karafInstallDir, File vmLoc) {
+		return String.format("%s%sjre%slib%sendorsed%s%s%slib%sendorsed%s%s%slib%sendorsed", vmLoc.getPath(), SEPARATOR, SEPARATOR, SEPARATOR, File.pathSeparator, vmLoc.getPath(),
+				SEPARATOR, SEPARATOR, File.pathSeparator, karafInstallDir, SEPARATOR, SEPARATOR);
+	}
+
```

Comment:
ok I will update that on Monday. I just extracted this code and didn't take care of it but effectively it will surely be more readable :-)